test/test_blocks_on_demand.py

```diff
@@ -0,0 +1,106 @@
+"""
+Test blocks on demand mode.
+"""
+
+from test.rpc.rpc_utils import gateway_call
+
+import requests
+
+from .account import invoke
+from .settings import APP_URL
+from .shared import (
+    ABI_PATH,
+    CONTRACT_PATH,
+    PREDEPLOY_ACCOUNT_CLI_ARGS,
+    PREDEPLOYED_ACCOUNT_ADDRESS,
+    PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+)
+from .util import ReturnCodeAssertionError, call, deploy, devnet_in_background
+
+
+@devnet_in_background(
+    *[
+        *PREDEPLOY_ACCOUNT_CLI_ARGS,
+        "--blocks-on-demand",
+    ]
+)
+def test_blocks_on_demand_invoke():
+    """Test deploy in blocks on demand mode"""
+    latest_block = gateway_call("get_block", blockNumber="latest")
+    genesis_block_number = latest_block["block_number"]
+    assert genesis_block_number == 0
+
+    deploy_info = deploy(CONTRACT_PATH, inputs=["0"])
+    should_fail_on_contract_call = False
+    try:
+        call(
+            function="get_balance",
+            address=deploy_info["address"],
+            abi_path=ABI_PATH,
+        )
+    except ReturnCodeAssertionError:
+        should_fail_on_contract_call = True
+    assert should_fail_on_contract_call is True
+
+    invoke(
```

Comment:
If the deploy tx is not considered either PENDING or ACCEPTED_ON_L2 at this point, does it make sense to be able to invoke? What is the behavior on alpha-goerli for this scenario?

Reply:
It took some time but I tested it and when the transaction is it's `Received` you can't `invoke, there is a `StarknetErrorCode.UNINITIALIZED_CONTRACT` error, I couldn't test the `PENDING` state since it's an instantly `ACCEPTED_ON_L2` after `Received`

When it's `ACCEPTED_ON_L2` you can `invoke` it.